modules/mvt/src/lib/geojson-vt/geojson-tiler.ts

```diff
@@ -0,0 +1,253 @@
+  indexMaxPoints?: number /** max number of points per tile in the tile index */;
+  tolerance?: number /** simplification tolerance (higher means simpler) */;
+  extent?: number /** tile extent */;
+  buffer?: number /** tile buffer on each side */;
+  lineMetrics?: boolean /** whether to calculate line metrics */;
+  promoteId?: string /** name of a feature property to be promoted to feature.id */;
+  generateId?: boolean /** whether to generate feature ids. Cannot be used with promoteId */;
+  debug?: number /** logging level (0, 1 or 2) */;
+};
+
+const defaultOptions: Required<GeoJSONTilerOptions> = {
+  maxZoom: 14, // max zoom to preserve detail on
+  indexMaxZoom: 5, // max zoom in the tile index
+  indexMaxPoints: 100000, // max number of points per tile in the tile index
+  tolerance: 3, // simplification tolerance (higher means simpler)
+  extent: 4096, // tile extent
+  buffer: 64, // tile buffer on each side
+  lineMetrics: false, // whether to calculate line metrics
+  // @ts-expect-error Ensures all these required params have defaults
+  promoteId: undefined, // name of a feature property to be promoted to feature.id
+  generateId: false, // whether to generate feature ids. Cannot be used with promoteId
+  debug: 0 // logging level (0, 1 or 2)
+};
+
+export class GeoJSONTiler {
+  options: Required<GeoJSONTilerOptions>;
+
+  // tiles and tileCoords are part of the public API
+  tiles: Record<string, any> = {};
+  tileCoords: any[] = [];
+
+  stats = {};
+  total = 0;
+
+  constructor(data, options?: GeoJSONTilerOptions) {
+    options = this.options = {...defaultOptions, ...options};
+
+    const debug = options.debug;
+
+    if (debug) console.time('preprocess data');
+
+    if (this.options.maxZoom < 0 || this.options.maxZoom > 24) {
+      throw new Error('maxZoom should be in the 0-24 range');
+    }
+    if (options.promoteId && options.generateId) {
+      throw new Error('promoteId and generateId cannot be used together.');
+    }
+
+    // projects and adds simplification info
+    let features = convert(data, options);
+
+    if (debug) {
+      console.timeEnd('preprocess data');
+      console.log(
+        'index: maxZoom: %d, maxPoints: %d',
+        options.indexMaxZoom,
+        options.indexMaxPoints
+      );
+      console.time('generate tiles');
+    }
+
+    // wraps features (ie extreme west and extreme east)
+    features = wrap(features, options);
+
+    // start slicing from the top tile down
+    if (features.length) {
+      this.splitTile(features, 0, 0, 0);
+    }
+
+    if (debug) {
+      if (features.length) {
+        console.log('features: %d, points: %d', this.tiles[0].numFeatures, this.tiles[0].numPoints);
+      }
+      console.timeEnd('generate tiles');
+      console.log('tiles generated:', this.total, JSON.stringify(this.stats));
+    }
+  }
+
+  // eslint-disable-next-line complexity
+  getTile(z: number, x: number, y: number) {
+    // z = +z;
+    // x = +x;
+    // y = +y;
```

Comment:
I guess this was to catch and allow `-0`?

Reply:
The test suite shows that this was to allow strings to be used instead of numbers. I chose not to support such "sloppy" overloading.